library/gen_837i.inc.php

```diff
@@ -0,0 +1,1946 @@
+    "~\n";
+
+  if ($CMS_5010 || ($claim->billingFacilityNPI() && $claim->billingFacilityETIN())) {
+    ++$edicount;
+    $out .= "REF" ;
+    if($claim->federalIdType()){
+      $out .= "*" . $claim->federalIdType();
+    }
+    else{
+      $out .= "*EI"; // For dealing with the situation before adding TaxId type In facility.
+    }
+    $out .=  "*" . $claim->billingFacilityETIN() .
+      "~\n";
+  }
+
+  if ($claim->providerNumberType() && $claim->providerNumber() &&
+      !($CMS_5010 && $claim->billingFacilityNPI()))
+  {
+    ++$edicount;
+    $out .= "REF" .
+      "*" . $claim->providerNumberType() .
+      "*" . $claim->providerNumber() .
+      "~\n";
+  }
+  else if ($claim->providerNumber() && !$claim->providerNumberType()) {
+    $log .= "*** Payer-specific provider insurance number is present but has no type assigned.\n";
+  }
+
+  // Situational PER*1C segment.
+  ++$edicount;
+  $out .= "PER" .
+    "*IC" .
+    "*" . $claim->billingContactName() .
+    "*TE" .
+    "*" . $claim->billingContactPhone();
+  if (!$CMS_5010 && $claim->x12gsper06()) {
+    $out .= "*ED*" . $claim->x12gsper06();
+  }
+  $out .= "~\n";
+
+  // Pay-To Address defaults to billing provider and is no longer required in 5010.
+  if (!$CMS_5010) {
+    ++$edicount;
+    // Field length is limited to 35. See nucc dataset page 63 www.nucc.org
+    $billingFacilityName = substr($claim->billingFacilityName(), 0, $CMS_5010 ? 60 : 35);
+    $out .= "NM1" .       // Loop 2010AB Pay-To Provider
+      "*87" .
+      "*2" .
+      "*" . $billingFacilityName .
+      "*" .
+      "*" .
+      "*" .
+      "*";
+    if ($claim->billingFacilityNPI())
+      $out .= "*XX*" . $claim->billingFacilityNPI();
+    else
+      $out .= "*24*" . $claim->billingFacilityETIN();
+    $out .= "~\n";
+
+    ++$edicount;
+    $out .= "N3" .
+      "*" . $claim->billingFacilityStreet() .
+      "~\n";
+
+    ++$edicount;
+    $out .= "N4" .
+      "*" . $claim->billingFacilityCity() .
+      "*" . $claim->billingFacilityState() .
+      "*" . stripZipCode($claim->billingFacilityZip()) .
+      "~\n";
+
+    if ($claim->billingFacilityNPI() && $claim->billingFacilityETIN()) {
+      ++$edicount;
+      $out .= "REF" .
+        "*EI" .
+        "*" . $claim->billingFacilityETIN() .
+        "~\n";
+    }
+  }
+
+  // This is also excluded in the 837I
+  // Loop 2010AC Pay-To Plan Name omitted.  Includes:
+  // NM1*PE, N3, N4, REF*2U, REF*EI
+
+  $PatientHL = $claim->isSelfOfInsured() ? 0 : 1;
+  $HLSubscriber = $HLcount++;
+
+  ++$edicount;
+  
+  // loop 2000B 
+  $out .= "HL" .        // Loop 2000B Subscriber HL Loop 
+    "*$HLSubscriber" .
+    "*$HLBillingPayToProvider" .
+    "*22" .
+    "*$PatientHL" .
+    "~\n";
+
+  if (!$claim->payerSequence()) {
+    $log .= "*** Error: Insurance information is missing!\n";
+  }
+
+  ++$edicount;
+  
+  //SBR01 is either a P or S    SBR02 for care is always 18 "patient"  SBR09 is always MA
```

Comment:
This is an institutional claim thing where you can't use a different subscriber?